File: pkg/models/message.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Message struct {
	ID        string         `json:"id" gorm:"primaryKey"`
	UserID    string         `json:"user_id"`  // aka the Author of the message
	GroupID   string         `json:"group_id"` // to filter the message into the right chat
	Content   string         `json:"content"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index"`
}

type UpdateMessage struct {
	Content string `json:"content"`
}

File: frontend/src/types/message.ts
export type MessageRequest = {
  group_id: string;
  content: string;
};

export type UpdateMessage = {
  content: string;
};

export type WsRefreshMessage = {
  new_group_id: string;
  clients_to_update: string[];
};

export type Message = {
  id: string;
  user_id: string;
  group_id: string;
  content: string;
  created_at: Date;
  updated_at: Date;
  DeletedAt: boolean | Date;
};
